Tasks/SonarQubePreBuild/SonarQubeHelper.ps1

```diff
@@ -1,3 +1,20 @@
+#
+# Remarks: Some sensitive parameters cannot be stored on the agent between the 2 steps so 
+# we'll store them in the task context and pass them to the post-test step
+#
+function StoreSensitiveParametersInTaskContext
+{ 
+	param(
+		  [string]$serverUsername,
+		  [string]$serverPassword,
+		  [string]$dbUsername,
+		  [string]$dbPassword)
+
+	SetTaskContextVaraible "serverUsername" $serverUsername
+	SetTaskContextVaraible "serverPassword" $serverPassword
+	SetTaskContextVaraible "dbUsername" $dbUsername
+	SetTaskContextVaraible "dbPassword" $dbPassword
+}
 
 function CreateCommandLineArgs
```

Comment:
Good point. We currently do not allow the user to NOT specify a generic endpoint, and in a generic endpoint you need to specify the server URL. Although there is an advanced scenario where you specify everything in the analysis.config, JM is of the opinion that a minimal set of settings should be specified via the task UI (i.e. the host url, the name, the version and the key).